### ISSUE TITLE
Support asterisks in version nuget constraints

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#26140]
### Describe the proposed change(s).

Example from https://github.com/renovate-reproductions/24921

```
DEBUG: Dependency Microsoft.Extensions.Configuration has unsupported/unversioned value 7.0.* (versioning=nuget) (repository=renovate-reproductions/24921)
DEBUG: Dependency Microsoft.Extensions.Configuration has unsupported/unversioned value 6.0.* (versioning=nuget) (repository=renovate-reproductions/24921)
DEBUG: Dependency Microsoft.Extensions.Configuration has unsupported/unversioned value 6.0.* (versioning=nuget) (repository=renovate-reproductions/24921)
DEBUG: Dependency Microsoft.Win32.Registry has unsupported/unversioned value 5.* (versioning=nuget) (repository=renovate-reproductions/24921)
DEBUG: Dependency Microsoft.Extensions.Configuration has unsupported/unversioned value 6.0.* (versioning=nuget) (repository=renovate-reproductions/24921)
DEBUG: Dependency Meziantou.Analyzer has unsupported/unversioned value 2.0.* (versioning=nuget) (repository=renovate-reproductions/24921)
```

The original request in https://github.com/renovatebot/renovate/discussions/24921 is for asterisks in suffixes too, e.g. `3.4.142-*`

**Comment by n00j** [comment]
Looks like the PR was merged in. Is this support now available?

**Comment by viceice** [comment]
@zharinov is it done?

**Comment by zharinov** [comment]
It's done

**Comment by n00j** [comment]
Does this also properly suggest upgrades to wild card versions? 

I.e. if I have a dependency on '0.1.*'  and a new nugget version is published '0.2.1'. Will a or be created to upgrade to the new version?

Thanks!
